account_payment_order/models/account_payment_order.py

```diff
@@ -433,7 +433,10 @@ def generated2uploaded(self):
             for line in payment.payment_line_ids:
                 if not line.move_line_id:
                     continue
-                if line.amount_currency != -line.move_line_id.amount_residual_currency:
+                sign = -1 if payment.payment_order_id.payment_type == "outbound" else 1
+                if line.amount_currency != (
```

Comment:
Should we take the opportunity to use `float_compare` here?

Reply:
@pedrobaeza Swapped